tests/gdbstub.sh

```diff
@@ -11,20 +11,31 @@ then
         -ex "file build/puzzle.elf" \
         -ex "target remote :1234" \
         -ex "break *0x10700" \
```

Comment:
Once you utilize bash array, the break commands can be generated accordingly.

Reply:
But the repeat pattern only exists for a part of command arguments. I am not sure how could I use the bash array on this.

Reply:
Check this: https://gist.github.com/mateuszalmannai/8bc8cff1e84318d82c11004f86057966
You can iterative each element in a bash array (index -> breakpoint address) and generate the GDB commands associated with breakpoint addresses.

Reply:
I am still confused about this. Since it looks like we should run the  `riscv32-unknown-elf-gdb` command once for the test, and the part we may want to loop on is only a few of the `-ex` arguments. We can't do something like this:

```
riscv32-unknown-elf-gdb --batch \
               for t ${allThreads[@]}; do
                   -ex "target remote :1234" \
               done
               ...
```

Reply:
Alternatively, you can simply operate as following
```shell
riscv32-unknown-elf-gdb --batch $(GDB_COMMANDS)
```

Here, `$(GDB_COMMANDS)` consists of generated sequences.